test/envoye2e/driver/grpc.go

```diff
@@ -74,3 +74,48 @@ func (g *GrpcCall) Run(p *Params) error {
 }
 
 func (g *GrpcCall) Cleanup() {}
+
+var _ Step = &GrpcStream{}
+
+type GrpcStream struct {
+	Counts []uint32
+}
+
+func (g *GrpcStream) Run(p *Params) error {
+	proxyAddr := fmt.Sprintf("127.0.0.1:%d", p.Ports.ClientPort)
+	conn, err := grpc.Dial(proxyAddr, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		return fmt.Errorf("could not establish client connection to gRPC server: %v", err)
+	}
+	defer conn.Close()
+	client := grpc_echo.NewEchoClient(conn)
+
+	stream, err := client.EchoStream(context.Background())
+	if err != nil {
+		return err
+	}
+
+	var count uint32
+	i := 0
+
+	for {
+		if i >= len(g.Counts) {
```

Comment:
Why not just do this and i++ in the top for loop?

Reply:
Simplified.